### PR TITLE
FIX-#4300: Fix Modin Categorical column dtype categories

### LINKED ISSUE #4275: Setting a column of a Modin DataFrame to Categorical twice does not work
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: macOS Version 11.6
- **Modin version** (`modin.__version__`): 0.13.2
- **Python version**: Python 3.8.11
- **Code we can use to reproduce**:

```
import modin.pandas as pd
from modin.pandas import Categorical
 
d = {'a': [1, 2, 3], 'b': [4, 5, 6]} 
df = pd.DataFrame(data=d)
df['a'] = Categorical(df['a'], ordered=True)
df['a'] = Categorical(df['a'], ordered=True)
```

### Describe the problem

This results in an `AttributeError: 'Series' object has no attribute '_codes'`.

### LINKED ISSUE #4300: Modin Categorical column categories are not correct.
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: MacOS Version 11.6
- **Modin version** (`modin.__version__`): 0.13.2
- **Python version**: 3.8.11
- **Code we can use to reproduce**:

```
import modin.pandas as pd
from modin.pandas import Categorical
 
d = {'a': [1, 2, 3], 'b': [4, 5, 6]} 
df = pd.DataFrame(data=d)
df['a'] = Categorical(df['a'], ordered=True)
df['a'].dtype
```

### Describe the problem

The categories for Modin only include the first category and are not a complete list when compared to pandas. The output of `CategoricalDtype(categories=[1], ordered=True)` in the code above in Modin vs. `CategoricalDtype(categories=[1, 2, 3], ordered=True)` in pandas.

### PR DESCRIPTION
**Pull request by naren-ponder**
Signed-off-by: Naren Krishna <naren@ponder.io>

## What do these changes do?

Fixes Modin Categorical dtype.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4300 ? 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [ ] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: modin/core/storage_formats/pandas/parsers.py [hunk 1]
```diff
@@ -123,13 +123,7 @@ def find_common_type_cat(types):
     """
     if all(isinstance(t, pandas.CategoricalDtype) for t in types):
         if all(t.ordered for t in types):
-            categories = np.unique([c for t in types for c in t.categories])[0]
-            if categories.ndim != 0:
-                categories = np.sort(categories)
-            else:
-                # Cannot use np.sort() on array with ndim == 0
```
[the remark this answers is not in the document]
Comment (naren-ponder) [reply modin/core/storage_formats/pandas/parsers.py:130]: The original code for this comes from https://github.com/modin-project/modin/pull/4258 (which is something I wrote to fix a reported bug). Turns out that the pandas functions around Categorical and the `categories.ndim == 0` case won't actually get hit. Even a category with a single scalar will pass with this change.

[the remark this answers is not in the document]
Comment (mvashishtha) [reply modin/core/storage_formats/pandas/parsers.py:130]: Then the new code LGTM.

**Review by mvashishtha — APPROVED** [review]
LGTM

**Review by devin-petersohn — APPROVED** [review]
LGTM, will merge when CI passes

**Pull request merged**